build/tools/esp32/esptool_py/Makefile

```diff
@@ -74,19 +74,32 @@ $(APP_BIN_UNSIGNED): $(APP_ELF) $(ESPTOOLPY_SRC)
 
 ESPTOOL_ALL_FLASH_ARGS += $(ESPTOOL_BOOTLOADER_FLASH_ARGS)
 
-ALL: flash romfs
+ESPTOOLPY_SUBMODULE := esptool_py/esptool
 
-OS:app-flash
+ALL: check-modules flash romfs
 
-ROMFS:romfs
+OS: check-modules app-flash
+
+ROMFS: check-modules romfs
+
+BOOTLOADER: check-modules bootloader
+
+ERASE_ALL: check-modules erase_flash
+
+check-modules:
```

Comment:
1. Do we need to make a new target and need to call it at every target? If this is mandatory for all of activities, how about removing the target?
2. When esptool.py is not existed, how about executing pulling submodule automatically through this file?
  If it is not good, how about gibing guidance how to get submodule?

Reply:
Yes, good suggestion. The new target is not a good way. 
Thanks.

Reply:
@sunghan-chang 
Dear Mr. Chang
  1.  I have checked the dependency of all the targets in this Makefile, and I think it's better to add a new target and make it to be the first dependency of all the targets, That's because all the target depends on this esptool.py file, so i change this dependency to the new target.
   2. The new target will get the esptool.py submodule automatically, and i have made the verification on both docker and local build. It works.

If you have any other suggestion, please inform me. Thanks.

Reply:
@sunghan-chang  
I have tried to define a function, but i need ifeq to check the exist of  esptool.py, it would call a issue. Because function will be called in target content and a tab is needed before calling this function as:
romfs:
[tab]$(call function)
It will call a error due to ifeq needs no blank before it, but well have a tab before calling functions in a target.
if we remove the ifeq to each target, it will bring in too much duplicate code. And also if we add the ifeq checking in every target, it would bring in new redundant operations, because targets have their dependency and the dependency will be executed firstly. 

So, I think a new target is better in simple code and less operations.